File: backend/src/domain/models/UserProgress.js
const mongoose = require('mongoose');

const userProgressSchema = new mongoose.Schema({
  _id: { type: String, ref: 'User', required: true }, 
  total_learned: { type: Number, default: 0 },
  streak_days: { type: Number, default: 0 },
  last_study_date: { type: Date, default: null }
}, { 
  timestamps: { createdAt: false, updatedAt: 'updated_at' } 
});

module.exports = mongoose.model('UserProgress', userProgressSchema);

File: android/app/src/main/java/com/example/android/domain/model/UserProgress.kt
package com.example.android.domain.model

data class UserProgress(
    val userId: String,
    val totalLearned: Int = 0,
    val streakDays: Int = 0,
    val lastStudyDate: Long? = null,
    val updatedAt: Long = System.currentTimeMillis()
)
